clients/drcachesim/tracer/physaddr.cpp

```diff
@@ -178,8 +186,11 @@ physaddr_t::virtual2physical(addr_t virt, OUT addr_t *phys)
         NOTIFY(1, "v2p failure: entry is invalid for %p\n", vpage);
         return false;
     }
-    last_ppage_ = (addr_t)((entry & PAGEMAP_PFN) << page_bits_);
+    // TODO i#4014: On recent kernels unprivileged reads succeed but are passed
+    // a 0 PFN.  Since that could be valid, we need to check our capabilities
+    // to decide.  Until then, we're currently returning 0 for every query.
```

Comment:
s/for every query/for every failed query/

Reply:
Used "for every unprivileged query" since "failed" can be confusing.